### PR TITLE
fix login api and doc after merge master

### PR DESCRIPTION
**Pull request by mvn-annguyen-dn**
## Description
fix structure login api after update portal
## Reference Requirements
- Product Backlog ID: 39
- Task ID: 39.3

## Reference Isuees
#{issue_id}

## Migrations
YES 

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

**Comment by at-vietpham** [comment]
LGTM 🐙

File: app/Http/Controllers/Api/LoginController.php [hunk 1]
```diff
@@ -21,44 +21,43 @@ class LoginController extends ApiController
...
     {
             $portalResponse = callAPIPortal($request);
-        if ($portalResponse->meta->code == Response::HTTP_OK) {
             $user = $this->saveUser($portalResponse, $request);
-                
-            return  response()->json([
-            'meta' => $portalResponse->meta,
-            'data' => $user,
-            ], Response::HTTP_OK);
+        if (isset($user['errors']['message'])) {
```
Comment (at-nhuibo) [inline app/Http/Controllers/Api/LoginController.php]: Check cái này trước rồi mới `$user = $this->saveUser($portalResponse, $request);` nha.

File: app/Http/Controllers/Api/LoginController.php [hunk 2]
```diff
@@ -21,44 +21,43 @@ class LoginController extends ApiController
...
-        $user = [
-            'name' => $userResponse->name,
-            'team' => $userResponse->teams[0]->name,
-            'expired_at' => $date,
-            'avatar_url' => $userResponse->avatar_url,
-            'access_token' => $userResponse->access_token,
-        ];
-        if ($userResponse->teams[0]->name == User::SA) {
-            $user['role'] = User::ROOT_ADMIN;
+        try {
```
Comment (at-nhuibo) [inline app/Http/Controllers/Api/LoginController.php]: Thêm transaction vào cái này nha.

File: app/Http/Controllers/Api/LoginController.php [hunk 3]
```diff
@@ -21,44 +21,43 @@ class LoginController extends ApiController
...
             $user = $this->saveUser($portalResponse, $request);
-                
-            return  response()->json([
-            'meta' => $portalResponse->meta,
-            'data' => $user,
-            ], Response::HTTP_OK);
+        if (isset($user['errors']['message'])) {
+            return metaResponse(null, $user['errors']['code'], $user['errors']['message']);
         }
+            return  metaResponse(['data' => $user]);
```
Comment (at-nhuibo) [inline app/Http/Controllers/Api/LoginController.php]: Thụt vào 1 tab nữa.

**Review by ghost — CHANGES_REQUESTED** [review]
Check review nhé!

**Review by ghost — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**